### PR TITLE
Add support for 'other' gender

### PR DESCRIPTION
**Pull request by fw42**
Closes https://github.com/fw42/cubecomp/issues/246.

### Before

screen shot 2018-03-04 at 13 53 49

### After

screen shot 2018-03-04 at 13 48 40

### CODE REVIEW

**Comment by fw42** [comment]
@jfly or @timhabermaas, mind taking a quick look?

File: config/locales/fr.yml [hunk 1]
```diff
@@ -96,9 +96,12 @@ fr:
     newcomer: "Novice"
     non-newcomer: "Non-novice"
 
-  misc:
+  genders:
     male: "masculin"
     female: "féminin"
+    other: 'autre'
```
Comment (fw42) [inline config/locales/fr.yml]: @SAuroux: does this sound correct in French?

File: config/locales/pt.yml [hunk 2]
```diff
@@ -97,9 +97,12 @@ pt:
   disclaimer:
     title: "Organização"
 
-  misc:
+  genders:
     male: 'masculino'
     female: 'feminino'
+    other: 'inny'
```
Comment (fw42) [inline config/locales/pt.yml]: @cubizh, does this sound correct in Portuguese?

File: config/locales/it.yml [hunk 3]
```diff
@@ -97,9 +97,12 @@ it:
   disclaimer:
     title: "Organizzazione"
 
-  misc:
+  genders:
     male: "Maschio"
     female: "Femmina"
+    other: 'altro'
```
Comment (fw42) [inline config/locales/it.yml]: I don't remember who I added the Italian translations for. Anyone speak Italian? @SAuroux, do you know anyone at the WCA that can confirm that this is the correct translation?

File: config/locales/pt.yml (hunk 2 above)
Comment (cubizh) [reply config/locales/pt.yml]: No, please change it to 'outro'

File: app/views/admin/competitors/index.html.erb [hunk 4]
```diff
@@ -64,9 +64,9 @@
           <% end %>
         </td>
         <td>
-          <% if competitor.male? %>
+          <% if competitor.gender == 'male' %>
```
Comment (timhabermaas) [inline app/views/admin/competitors/index.html.erb]: I'd use `competitor.male?`/`competitor.female?` to hide the actual value representing male/female.

File: db/migrate/20180304131900_competitors_add_gender_column.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class CompetitorsAddGenderColumn < ActiveRecord::Migration[5.1]
+  def change
+    add_column :competitors, :gender, :string
```
Comment (timhabermaas) [inline db/migrate/20180304131900_competitors_add_gender_column.rb]: `null: false`?

File: db/migrate/20180304132024_competitors_migrate_gender_column.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  def change
+    migrate_competitors(male: true, gender: 'male')
+    migrate_competitors(male: false, gender: 'female')
+  end
+
+  private
+
+  def migrate_competitors(male:, gender:)
+    Competitor.where(male: male).find_in_batches do |competitors|
+      Competitor.where(id: competitors.map(&:id)).update_all(gender: gender)
```
Comment (jfly) [inline db/migrate/20180304132024_competitors_migrate_gender_column.rb]: This feels a bit weird to me.  Can't you do this in one SQL command instead?

```ruby
Competitor.where(male: male).update_all(gender: gender)
```

Is the idea to give your sql server a break during this migration?

Comment (fw42) [reply db/migrate/20180304132024_competitors_migrate_gender_column.rb]: Yea, total premature optimisation for this tiny database, lol

**Comment by fw42** [comment]
Thanks for the translations, Jeremy, that's great! 👍

**Pull request merged**